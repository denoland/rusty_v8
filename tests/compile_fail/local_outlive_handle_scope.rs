fn main() {
  let isolate = &mut v8::Isolate::new(Default::default());
  v8::scope!(let scope1, isolate);

  let _local = {
    v8::scope!(let scope, scope1);

    v8::Integer::new(scope, 123)
  };
}
