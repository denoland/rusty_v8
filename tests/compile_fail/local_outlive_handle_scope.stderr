error[E0597]: `scope` does not live long enough
 --> tests/compile_fail/local_outlive_handle_scope.rs:6:5
  |
5 |   let _local = {
  |       ------ borrow later stored here
6 |     v8::make_handle_scope!(let scope, scope1);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     borrowed value does not live long enough
  |     binding `scope` declared here
...
9 |   };
  |   - `scope` dropped here while still borrowed
  |
  = note: this error originates in the macro `$crate::make_handle_scope` which comes from the expansion of the macro `v8::make_handle_scope` (in Nightly builds, run with -Z macro-backtrace for more info)
